src/core/stream/adaptation/adaptation_stream.ts

```diff
@@ -359,7 +360,8 @@ export default function AdaptationStream<T>({
                                     segmentFetcher,
                                     terminate$: terminateCurrentStream$,
                                     bufferGoal$,
-                                    fastSwitchThreshold$ })
+                                    fastSwitchThreshold$,
+                                    getCurrentTime })
```

Comment:
This may need further debate but isn't this a little ugly?
As I read that code, I prefer more and more a function like `clockTick.getPrecizeMediaTime()`.